libgeos/libgeos_handle.go

```diff
@@ -0,0 +1,236 @@
+package libgeos
+
+/*
+#cgo linux CFLAGS: -I/usr/include
+#cgo linux LDFLAGS: -L/usr/lib -lgeos_c
+#include "geos_c.h"
+#include <stdlib.h>
+#include <string.h>
+
+void sf_error_handler(const char *message, void *userdata) {
+	strncpy(userdata, message, 1024);
+}
+
+GEOSContextHandle_t sf_init(void *userdata) {
+	GEOSContextHandle_t ctx = GEOS_init_r();
+	GEOSContext_setErrorMessageHandler_r(ctx, sf_error_handler, userdata);
+	return ctx;
+}
+
+*/
+import "C"
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"strings"
+	"unsafe"
+
+	"github.com/peterstace/simplefeatures/geom"
+)
+
+// Handle is an opaque handle that can be used to invoke libgeos operations.
+// Instances are not threadsafe and thus should not be used in a concurrent
+// manner.
```

Comment:
I think it would be better to write this comment as "Instances are not threadsafe and thus should only be used within some mutex-protected manner."

Reply:
Agreed, that's a bit clearer. I've switched the wording to say how it _should_ be used rather than how it _shouldn't_ be used (I didn't use your exact wording, but I think that's the distinction you were getting at).